client/modules/IDE/components/SketchList.jsx

```diff
@@ -25,14 +26,19 @@ class SketchList extends React.Component {
     return `p5.js Web Editor | ${this.props.username}'s sketches`;
   }
 
+  renderLoader() {
+    return this.props.loading && !this.props.sketches.length && <Loader />;
+  }
+
   render() {
     const username = this.props.username !== undefined ? this.props.username : this.props.user.username;
     return (
       <div className="sketches-table-container">
         <Helmet>
           <title>{this.getSketchesTitle()}</title>
         </Helmet>
-        { this.props.sketches.length === 0 &&
+        {this.renderLoader()}
+        { !this.props.loading && this.props.sketches.length === 0 &&
```

Comment:
last small thing—could you turn this into another helper method, which you could also call in `renderLoader()`?